src/overpass_api/statements/area_query.cc

```diff
@@ -287,8 +287,20 @@ void Area_Query_Statement::get_ranges
 
 bool Area_Constraint::delivers_data(Resource_Manager& rman)
 {
+  int counter = 0;
```

Comment:
`int counter = 0` in line 311 should probably be removed as well, otherwise line 321 won't return the correct value I guess (variable `counter` gets shadowed).